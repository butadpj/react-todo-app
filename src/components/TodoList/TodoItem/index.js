import React from 'react';

const TodoItem = (props) => {
  const todoNameBasedOnState = (todo) => {
    return todo.isDone ? (
      <p className='isDone'>{`${todo.name} âœ…`}</p>
    ) : (
      <p>{todo.name}</p>
    );
  };

  return (
    <li>
      {todoNameBasedOnState(props.todo)}
      <div className='todo__buttons'>
        <button
          className={`btn btn--primary toggle-btn ${
            props.todo.isDone ? 'undo-btn' : 'done-btn'
          }`}
          onClick={() => props.toggleIsDone(props.todo)}
        >
          {props.todo.isDone ? 'Undo' : 'Done'}
        </button>
        <button
          className='btn delete-btn'
          onClick={() => props.deleteTodo(props.todo.id)}
        >
          Delete
        </button>
      </div>
    </li>
  );
};

export default TodoItem