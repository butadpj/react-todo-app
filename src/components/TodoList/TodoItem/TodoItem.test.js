import TodoItem from "./index";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

const sampleTodo = { id: 1, name: 'Sample Todo', isDone: false };

describe('<TodoItem />', () => {
  it("renders without crashing", () => {
    shallow(<TodoItem todo={sampleTodo}/>);
  });

  it("UI doesn't change unexpectedly", () => {
    const wrapper = mount(<TodoItem todo={sampleTodo}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("Todo name is changing based on todo.isDone property", () => {
    const notCompletedTodo = mount(<TodoItem todo={{...sampleTodo, isDone: false}}/>);
    expect(notCompletedTodo.find('p').text()).toBe(sampleTodo.name);

    const completedTodo = mount(<TodoItem todo={{...sampleTodo, isDone: true}}/>);
    expect(completedTodo.find('p').text()).toBe(`${sampleTodo.name} âœ…`);
  });

  it("Toggle button's name is changing based on todo.isDone property", () => {
    const notCompletedTodo = mount(<TodoItem todo={{...sampleTodo, isDone: false}}/>);
    expect(notCompletedTodo.find('.toggle-btn').text()).toBe('Done');

    const completedTodo = mount(<TodoItem todo={{...sampleTodo, isDone: true}}/>);
    expect(completedTodo.find('.toggle-btn').text()).toBe('Undo');
  });

  it("Toggle button's className is changing based on todo.isDone property", () => {
    const notCompletedTodo = mount(<TodoItem todo={{...sampleTodo, isDone: false}}/>);
    expect(notCompletedTodo.find('.toggle-btn').hasClass('done-btn')).toBe(true);

    const completedTodo = mount(<TodoItem todo={{...sampleTodo, isDone: true}}/>);
    expect(completedTodo.find('.toggle-btn').hasClass('undo-btn')).toBe(true);
  });
});