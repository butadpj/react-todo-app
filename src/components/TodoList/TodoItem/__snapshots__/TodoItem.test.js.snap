// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoItem /> UI doesn't change unexpectedly 1`] = `
<TodoItem
  todo={
    Object {
      "id": 1,
      "isDone": false,
      "name": "Sample Todo",
    }
  }
>
  <li>
    <p>
      Sample Todo
    </p>
    <div
      className="todo__buttons"
    >
      <button
        className="btn btn--primary toggle-btn done-btn"
        onClick={[Function]}
      >
        Done
      </button>
      <button
        className="btn delete-btn"
        onClick={[Function]}
      >
        Delete
      </button>
    </div>
  </li>
</TodoItem>
`;
